src/redux/rockets/rocketsSlice.js

```diff
@@ -0,0 +1,40 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', async () => {
+  try {
+    const response = await axios.get('https://api.spacexdata.com/v4/rockets');
+    return response.data;
+  } catch (error) {
+    throw Error('Failed to fetch rocket data');
+  }
+});
+
+const initialState = {
+  rockets: [],
+  status: 'idle',
+  error: null,
+};
+
+const rocketSlice = createSlice({
+  name: 'rockets',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchRockets.pending, (state) => {
+        state.status = 'loading';
+        state.error = null;
+      })
+      .addCase(fetchRockets.fulfilled, (state, action) => {
+        state.status = 'succeeded';
+        state.rockets = action.payload;
+      })
+      .addCase(fetchRockets.rejected, (state, action) => {
+        state.status = 'failed';
+        state.error = action.error.message;
+      });
+  },
+});
```

Comment:
Here we could just left all the logic and rendering for components, for exmaple (state.Loading and state.error):

![image](https://github.com/victormc13/Space-Travelers-Hub/assets/110580299/a26706ed-8ba0-4424-9374-9103af6bd5b5)

finally in our component we can use conditional rendering like this:

![image](https://github.com/victormc13/Space-Travelers-Hub/assets/110580299/4588090e-385b-4069-a516-8ad0a92d963a)

Reply:
By using these specific string values for the status property, your Redux reducers can easily convey the current state of asynchronous operations, making it clearer how the application is handling these operations and rendering UI components accordingly.

For further information, please refer to the [official documentation.
](https://redux-toolkit.js.org/api/createAsyncThunk)